### ISSUE TITLE
Temporary file breaks stored directory opening on iOS.

### ISSUE THREAD

**Issue by zhelenskiy** [MohamedRejeb/Calf#90]
The change introduced by 0.4.1 after 0.4.0 breaks the following use case:
* I open a file
* Show the file name and a button to open the containing directory
* Open the directory with the button.
  ```kotlin
    val url: NSURL = NSURL.URLWithString("shareddocuments://$folder")?: return
    val canOpenUrl = UIApplication.sharedApplication.canOpenURL(url)
    if (canOpenUrl) {
        UIApplication.sharedApplication.openURL(url)
    }
  ```

Expected behaviour (0.4.0):
* The filename is real.
* The folder contains the file.
<img width="339" alt="image" src="https://github.com/MohamedRejeb/Calf/assets/55230817/e9f76266-b063-4743-bda5-d267fd977c3d">

<img width="565" alt="image" src="https://github.com/MohamedRejeb/Calf/assets/55230817/7161b0e8-851d-4034-818d-afd6ef20b756">

Actual behaviour:
* Fake filename.
* Opens non-existing folder.
<img width="335" alt="image" src="https://github.com/MohamedRejeb/Calf/assets/55230817/93cc71ad-8ad1-49ad-a9a0-a9d98e7d8705">
<img width="565" alt="image" src="https://github.com/MohamedRejeb/Calf/assets/55230817/e3eba258-23e4-4d8a-9b5d-9f113864a12b">

**Comment by MohamedRejeb** [comment]
Hi,
Thanks for opening this issue, maybe keeping a reference to the original path in KmpFile for iOS should fix this?

**Comment by zhelenskiy** [comment]
By the way why do you need to create a temporary file?

**Comment by MohamedRejeb** [comment]
Here's the full discussion: #54 
I'm using iOS `PHPicker` and it return an `NSURL` valid only inside the picker lambda so you can't keep a reference of the file and use it later to upload or something, that's why we need the temporary file so we can keep using it later.
It's possible to store the `NSData` instead of `NSURL` but it's not ideal, we only need to get `NSData` or `ByteArray` when we have to, we shouldn't keep it in the memory for so long.

**Comment by zhelenskiy** [comment]
How am I expected to save to the original file then?

**Comment by MohamedRejeb** [comment]
As I said, I will add a reference to the original file path for iOS: `KmpFile.originalPath` something like that.

**Comment by MohamedRejeb** [comment]
This should be fixed in `0.5.0`, you don't need to change anything in your code.
